java-generator/gradle-plugin/src/main/java/io/fabric8/java/generator/gradle/plugin/JavaGeneratorPluginExtension.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator.gradle.plugin;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.provider.Property;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Map;
+
+public abstract class JavaGeneratorPluginExtension {
+
+  public static final String NAME = "javaGen";
+  private Project gradleProject;
+
+  public JavaGeneratorPluginExtension(Project gradleProject) {
+    this.gradleProject = gradleProject;
+  }
+
+  /**
+   * The input file or directory to be used for generating sources
+   *
+   */
+  public abstract RegularFileProperty getSource();
+
+  public File getSourceOrDefault() {
+    return this.getSource().getAsFile().getOrElse(null);
+  }
+
+  /**
+   * The URLs to be used to download CRDs from remote locations
+   */
+  public abstract Property<String[]> getUrls();
+
+  public String[] getUrlsOrDefault() {
+    return this.getUrls().getOrElse(new String[] {});
+  }
+
+  /**
+   * The Download target folder for CRDs downloaded from remote URLs
+   *
+   */
+  public abstract DirectoryProperty getDownloadTarget();
+
+  public File getDownloadTargetOrDefault() {
+    return this.getDownloadTarget().getAsFile()
+        .getOrElse(this.gradleProject.getLayout().getProjectDirectory().dir("tmp").dir("input").getAsFile());
```

Comment:
is it "gradle idiomatic" to create a `tmp` folder at the root of the project?
I'm not sure I have seen this pattern before ...

Reply:
No, you're right, thanks for pointing this out. We could set this default differenlty. I'll have a look.

Reply:
thanks 👍

Reply:
This should be fixed now.